vendor/bat-native-ads/src/bat/ads/internal/ads_impl.cc

```diff
@@ -712,14 +757,85 @@ void AdsImpl::CachePageScore(
   }
 }
 
+bool AdsImpl::ShoppingSitesMatch(
+    const std::string& url,
+    std::map<std::string, std::vector<std::string>> shopping_site_categories) {
```

Comment:
Please change `std::map<std::string, std::vector<std::string>>` to `std::map<std::string, std::vector<std::string>>&`